students/Thai_H/lesson3/mail_room.py

```diff
@@ -0,0 +1,122 @@
+        M) Back to MAIN MENU\n\
+        Q) Quit the program\n\
+        Please type L, N, or B: ')
+    return user_selection
+#-------------------------------------------------------------------------
+
+def thank_you_letter():
+    while True:
+        thank_you_selection = thank_you_menu_selection()
+
+        if thank_you_selection.upper() == 'L':
+            create_report()
+        elif thank_you_selection.upper() == 'T':
+            send_thank_you_add_new_donation()
+        elif thank_you_selection.upper() == 'M':
+            main()
+        elif thank_you_selection.upper() == 'Q':
+            exit_program()
+        else:
+            print('\n\nNot a valid option!\n\n')
+#--------------------------------------------------------------------------
+
+
+def sort_donor_list(unsorted_list):
+    sorted_donation_list = []
+    for d in unsorted_list:
+        total_donation =  float(sum(d[1]))
+        number_of_donation = int(len(d[1]))
+        average = total_donation / number_of_donation
+        sorted_donation_list.append([d[0], total_donation, number_of_donation, average])
+    #
+    sorted_donation_list.sort(key = sort_by_total_donation, reverse = True)
+
+    return sorted_donation_list
+
+def sort_by_total_donation(sorted_donation_list):
+    return sorted_donation_list[1]
+
+
+#------------------------------------------------------------------------------
+def create_report():
+    sorted_list = sort_donor_list(list_of_donors)
+    report_header = ['Donor Name', 'Total Donation Amt', 'Num of Donation',  'Avg Per Donation']
+    print('\n\n{:<25}{:>20}{:>20}{:>20}'.format(*report_header))
+    print('-'*90)
+    for donor in sorted_list:
+        print('{:<26}${:>15,.2f}{:>15}{:>25,.2f}'.format(donor[0], donor[1], donor[2], donor[1] / donor[2]))
+    print('\n\n')
+#------------------------------------------------------------------------------
+def init_donor_list():
+    # create initial list of donors and their donations
+    global list_of_donors
+    list_of_donors = [
+                ['William Boeing', [12.86, 2.00, 3.00, 4.00, 5.00]],
+                ['Steve Jobs', [6.00, 7.00, 8.00]],
+                ['Paul Allen', [9.00, 10.00, 11.00]],
+                ['Charles Flint', [12.00, 13.00, 14.00]],
+                ['Thomas Edison', [15.00, 16.00, 17.00]]
+            ]
+#------------------------------------------------------------------------------
+def menu_selection():
+    response = input('\n\nMAIN MENU\nChoose from one of the options below:\n\
+        T) THANK-YOU MENU\n\
+        R) Create a Report\n\
+        Q) Quit the program\n\
+        Please type T, R, or Q: ')
+    return response
+#------------------------------------------------------------------------------
+def exit_program():
+    print('Bye!')
+    sys.exit()
+#------------------------------------------------------------------------------
+def main():
+    while True:
+        init_donor_list()
+        menu_action = menu_selection()
```

Comment:
you can do:
`menu_action = menu_selection().upper()` so you don't have to call `upper` multiple times